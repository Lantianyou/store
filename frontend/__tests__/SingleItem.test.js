import { mount } from 'enzyme'
import toJSON from 'enzyme-to-json'
import wait from 'waait'
import SingleItem, { SINGLE_ITEM_QUERY } from '../components/SingleItem'
import { MockedProvider } from 'react-apollo/test-utils'
import { fakeItem } from '../lib/testUtils'

describe('<SingleItem />', () => {
  it('renders with proper data', async () => {
    const mocks = [
      {
        request: { query: SINGLE_ITEM_QUERY, variables: { id: 'abc123' } },
        result: {
          data: {
            item: fakeItem()
          }
        }
      }
    ]
    const wrapper = mount(
      <MockedProvider mocks={mocks}>
        <SingleItem id="abc123" />
      </MockedProvider>
    )
    expect(wrapper.text()).toBe('loading...')
    await wait()
    wrapper.update()

    expect(toJSON(wrapper.find('h2'))).toMatchSnapshot()
    expect(toJSON(wrapper.find('p'))).toMatchSnapshot()
    expect(toJSON(wrapper.find('img'))).toMatchSnapshot()
  })

  it('errors', async () => {
    const mocks = [
      {
        request: { query: SINGLE_ITEM_QUERY, variables: { id: 'abc123' } },
        result: {
          errors: [{ message: 'Image not found' }]
        }
      }
    ]
    const wrapper = mount(
      <MockedProvider mocks={mocks}>
        <SingleItem id="abc123" />
      </MockedProvider>
    )
    await wait()
    wrapper.update()
    const item = wrapper.find('[data-test="graphql-error"]')
    expect(item.text()).toContain('Image not found')
    expect(toJSON(item)).toMatchSnapshot()
  })
})
