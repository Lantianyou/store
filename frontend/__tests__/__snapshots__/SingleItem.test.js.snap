// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SingleItem /> errors 1`] = `
<p
  data-test="graphql-error"
>
  <strong>
    Shoot!
  </strong>
  Image not found
</p>
`;

exports[`<SingleItem /> renders with proper data 1`] = `
<h2>
  Viewing 
  dogs are best
</h2>
`;

exports[`<SingleItem /> renders with proper data 2`] = `
<p>
  dogs
</p>
`;

exports[`<SingleItem /> renders with proper data 3`] = `
<img
  alt="dogs are best"
  src="dog.jpg"
/>
`;
